lib/diaspora_federation/entities/signed_retraction.rb

```diff
@@ -1,9 +1,28 @@
 module DiasporaFederation
   module Entities
+    # this entity represents a claim of deletion of a previously federated entity
+    #
+    # @see Validators::SignedRetractionValidator
     class SignedRetraction < Entity
+      # @!attribute [r] target_guid
+      #   guid of a post to be deleted
+      #   @see HCard#guid
+      #   @return [String] target guid
       property :target_guid
+
+      # @!attribute [r] target_type
+      #   @return [String] target type
       property :target_type
+
+      # @!attribute [r] sender_id
+      #   The diaspora ID of the person who deletes a post
+      #   @see Person#diaspora_id
+      #   @return [String] diaspora ID
       property :sender_id, xml_name: :sender_handle
+
+      # @!attribute [r] author_signature
+      #   Contains a signature of the entity using the private key of the author of a post
```

Comment:
The retractions are issued from [here](https://github.com/diaspora/diaspora/blob/develop/app/models/user.rb#L377-L383).

RelayableRetraction is used for relayables, SignedRetraction is used for posts, and Retraction is used for everything else, which is ATM includes only photo.